src/index.ts

```diff
@@ -746,7 +746,32 @@ const handleFilterOperator = <
 			filterStr = `${operator.slice(1)}(${alias}:${filterStr})`;
 			return addParentKey(filterStr, parentKey, '/');
 		}
-		// break
+
+		// TODO: if $canAccess the immediate parentKey will have '/canAccess()' appended to it
+		case '$canAccess': {
```

Comment:
I like `/device?$filter=application/Auth.canAccess()` and afaict from the test cases it should work since it seems that you can apply functions on properties
```
  - Name: Functions in commonExpr - singleNavigation
    Rule: commonExpr
    Input: Products/Model.BestProduct()/Name/Model.Available()

  - Name: $root with bound function call
    Rule: odataRelativeUri
    Input: Suppliers?$filter=$root/Sales/Special.MostPopularCategory(Where=ZipCode) eq 'Food'
```
so I expect that we can also do that inside a $filter on normal navigation resources as well.
See: https://github.com/oasis-tcs/odata-abnf/blob/main/abnf/odata-abnf-testcases.yaml#L2243-L2245
This allow seems to be allowed by the following rule
See: https://docs.oasis-open.org/odata/odata/v4.01/os/abnf/odata-abnf-construction-rules.txt


An alternative that I saw in the test cases (but didn't like as much) was:
```
  - Name: $root with unbound function call
    Rule: odataRelativeUri
    Input: Suppliers?$filter=Special.MostPopularCategory(What=$root/Sales,Where=ZipCode) eq 'Food'

  - Name: 4.16 Addressing All Entities in a Service - with type cast
    Rule: odataRelativeUri
    Input: $all/Model.Customer?$filter=contains(Name,'red')
```

Reply:
`/device?$filter=application/Auth.canAccess()` seems to be allowed by the following syntactic rules:
```
filter
-> "$filter" EQ boolCommonExpr
-> "$filter" EQ commonExpr
-> "$filter" EQ firstMemberExpr
-> "$filter" EQ memberExpr
-> "$filter" EQ directMemberExpr
-> "$filter" EQ propertyPathExpr
-> "$filter" EQ entityColNavigationProperty    [ collectionNavigationExpr ]
-> "$filter" EQ entityColNavigationProperty    [ collectionPathExpr ]
-> "$filter" EQ entityColNavigationProperty    "/" boundFunctionExpr
-> "$filter" EQ entityColNavigationProperty    "/" functionExpr
-> "$filter" EQ entityColNavigationProperty    "/" entityColFunction    functionExprParameters
-> "$filter" EQ odataIdentifier    "/" odataIdentifier    OPEN CLOSE


here are the rules used in details


filter = ( "$filter" / "filter" ) EQ boolCommonExpr

boolCommonExpr = commonExpr ; resulting in a Boolean

commonExpr = ( primitiveLiteral
             / arrayOrObject
             / rootExpr
             / firstMemberExpr <----
             / functionExpr
             / negateExpr 
             / methodCallExpr 
             / parenExpr 
             / castExpr 
             / isofExpr
             / notExpr
             ) 
             [ addExpr 
             / subExpr 
             / mulExpr 
             / divExpr
             / divbyExpr 
             / modExpr
             ]  
             [ eqExpr 
             / neExpr 
             / ltExpr  
             / leExpr  
             / gtExpr 
             / geExpr 
             / hasExpr 
             / inExpr 
             ]
             [ andExpr 
             / orExpr 
             ] 

firstMemberExpr = memberExpr <----
                / inscopeVariableExpr [ "/" memberExpr ]

memberExpr = directMemberExpr <----
           / ( optionallyQualifiedEntityTypeName / optionallyQualifiedComplexTypeName ) "/" directMemberExpr

directMemberExpr = propertyPathExpr <----
                 / boundFunctionExpr 
                 / annotationExpr

propertyPathExpr = ( entityColNavigationProperty [ collectionNavigationExpr ] <----
                   / entityNavigationProperty    [ singleNavigationExpr ]
                   / complexColProperty          [ complexColPathExpr ]
                   / complexProperty             [ complexPathExpr ] 
                   / primitiveColProperty        [ collectionPathExpr ]
                   / primitiveProperty           [ primitivePathExpr ]
                   / streamProperty              [ primitivePathExpr ]
                   )


entityColNavigationProperty = odataIdentifier

collectionNavigationExpr = [ "/" optionallyQualifiedEntityTypeName ]
                           ( collectionPathExpr <----
                           / keyPredicate [ singleNavigationExpr ] 
                           / filterExpr [ collectionNavigationExpr ]
                           )

collectionPathExpr = count [ OPEN expandCountOption *( SEMI expandCountOption ) CLOSE ]
                   / filterExpr [ collectionPathExpr ]
                   / "/" anyExpr
                   / "/" allExpr
                   / "/" boundFunctionExpr <----
                   / "/" annotationExpr

boundFunctionExpr = functionExpr ; boundFunction segments can only be composed if the type of the    
                                 ; previous segment matches the type of the first function parameter

functionExpr = [ namespace "." ]
               ( entityColFunction    functionExprParameters [ collectionNavigationExpr ] <----
               / entityFunction       functionExprParameters [ singleNavigationExpr ] 
               / complexColFunction   functionExprParameters [ complexColPathExpr ]
               / complexFunction      functionExprParameters [ complexPathExpr ] 
               / primitiveColFunction functionExprParameters [ collectionPathExpr ] 
               / primitiveFunction    functionExprParameters [ primitivePathExpr ] 
               )

entityColFunction    = odataIdentifier

functionExprParameters = OPEN [ functionExprParameter *( COMMA functionExprParameter ) ] CLOSE
```


See: https://docs.oasis-open.org/odata/odata/v4.01/os/abnf/odata-abnf-construction-rules.txt